.github/shared/src/array.js

```diff
@@ -20,3 +20,14 @@ export async function filterAsync(array, asyncPredicate) {
 export async function mapAsync(array, asyncMapper) {
   return Promise.all(array.map(asyncMapper));
 }
+
+/**
+ * @template T,U
+ * @param {T[]} array
+ * @param {(item: T, index: number, array: T[]) => Promise<U[]>} asyncMapper
+ * @returns {Promise<U[]>}
+ */
+export async function flatMapAsync(array, asyncMapper) {
```

Comment:
I ended up not needing this helper, but we'll need it soon enough, so let's just add it now.